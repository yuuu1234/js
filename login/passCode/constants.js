export const PASSCODE_MAX_LENGTH = 6;

export default {
  PASSCODE_MAX_LENGTH,
};
