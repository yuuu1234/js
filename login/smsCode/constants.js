import numeral from 'numeral';

export const COUNTDOWN_TEXT = 'Code Expires in';
export const UNITTEXT = 'secs';
export const SECONDS_LIMIT = 60;
export const NUMBER_OF_DIGITS = 6;
export const TITLE_USER_SMS_INPUT_SCREEN = 'We Are Sending You An SMS Code';
export const FIREBASE_ERROR_MESSAGES = {
  'auth/invalid-phone-number': 'The phone number is too long.',
  'auth/invalid-verification-code': 'The SMS code is invalid. Please double check or resend it.',
  default: 'There has been an error processing your request.',
};

export function isSmsCodeValid(code, length) {
  return code && code.length === length && code === numeral(code).value().toString();
}
