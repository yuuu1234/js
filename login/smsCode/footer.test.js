import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import * as navigatorConstants from '../../../constants/navigatorConstants';
import Footer from './footer';

describe('Footer', () => {
  const props = {
    handleNavigation: jest.fn(),
    loginStore: {
      verifyPhoneNumber: jest.fn().mockReturnValue(Promise.resolve({ isVerified: true })),
      clearSMSCode: jest.fn(),
      smsCodeValidation: true,
    },
  };
  const component = <Footer {...props} />;
  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe('CountdownTimer', () => {
    it('should render CountdownTimer by default', () => {
      const countDownTimer = wrapper.find('CountdownTimer');

      expect(countDownTimer.exists()).toBeTruthy();
    });

    it('should render resend code area if isResendCode is true', () => {
      const countDownTimer = wrapper.find('CountdownTimer');
      let resendCode = wrapper.find({ testID: 'resend-code' });
      countDownTimer.props().onTimesUp();
      expect(resendCode.exists()).toBeFalsy();

      wrapper.update();
      resendCode = wrapper.find({ testID: 'resend-code' });
      expect(resendCode.exists()).toBeTruthy();
    });

    it('should hide resend code area if user click resend', () => {
      const countDownTimer = wrapper.find('CountdownTimer');
      let resendCode = wrapper.find({ testID: 'resend-code' });
      countDownTimer.props().onTimesUp();
      wrapper.update();
      resendCode = wrapper.find({ testID: 'resend-code' }).at(0);
      resendCode.simulate('click');
      wrapper.update();
      resendCode = wrapper.find({ testID: 'resend-code' }).at(0);

      expect(resendCode.exists()).toBeFalsy();
      expect(countDownTimer.exists()).toBeTruthy();
    });
  });

  describe('handleVerifyCode', () => {
    it('should navigate to phone number login page if smsCodeValidation && res.isVerified', async () => {
      const verifyButton = wrapper.find('FormButton');
      await verifyButton.simulate('click');

      expect(props.loginStore.clearSMSCode).toHaveBeenCalled();
      expect(props.handleNavigation).toHaveBeenCalledWith(navigatorConstants.PhoneNumberLogin);
    });

    it('should do nothing if res.isVerified is false', async () => {
      wrapper.setProps({
        loginStore: {
          verifyPhoneNumber: jest.fn().mockReturnValue(Promise.resolve({ isVerified: false })),
          clearSMSCode: jest.fn(),
          smsCodeValidation: true,
        },
      });
      wrapper.update();
      const verifyButton = wrapper.find('FormButton');
      await verifyButton.simulate('click');

      expect(props.loginStore.clearSMSCode).not.toHaveBeenCalled();
      expect(props.handleNavigation).not.toHaveBeenCalled();
    });
  });
});
