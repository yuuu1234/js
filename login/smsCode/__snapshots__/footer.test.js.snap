// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render component correctly 1`] = `
<View>
  <View
    style={
      Object {
        "paddingBottom": 50,
        "textAlign": "center",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
        }
      }
    >
      <Text>
        Code Expires in 
      </Text>
      <Text
        style={
          Object {
            "color": "#003D9A",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        60 
      </Text>
      <Text
        style={
          Object {
            "color": "#003D9A",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        secs
      </Text>
    </View>
  </View>
  Object {
    "children": Array [
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Verify
      </Text>,
    ],
    "props": Object {
      "accessibilityLabel": "FormButton",
      "accessible": true,
      "focusable": true,
      "onClick": [Function],
      "onResponderGrant": [Function],
      "onResponderMove": [Function],
      "onResponderRelease": [Function],
      "onResponderTerminate": [Function],
      "onResponderTerminationRequest": [Function],
      "onStartShouldSetResponder": [Function],
      "style": Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#003D9A",
          "borderRadius": 15,
          "height": 41,
          "justifyContent": "center",
          "marginBottom": 16,
          "marginHorizontal": 49,
          "marginTop": 6,
        },
        false,
        Object {},
      ],
    },
    "type": "View",
  }
</View>
`;
